### PR TITLE
70 create the become driver request

### PR DESCRIPTION
**Pull request by danube20**
Board card: #70

### CODE REVIEW

File: app/models/driver_request.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  validates :driver_licence_front, presence: true
+  validates :driver_licence_back, presence: true
+
+  enum stage: { requested: 0, accepted: 10, rejected: 20 }, _default: 0
+
+  scope :by_user, ->(user) { where("user_id = ?", user) }
+  scope :by_requested, ->() { where("stage = 0") }
+  scope :by_accepted, ->() { where("stage = 10") }
+  scope :by_rejected, ->() { where("stage = 20") }
```
Comment (yauhenininjia) [inline app/models/driver_request.rb:14]:
```ruby
scope :by_user, ->(user) { where(user_id: user.id) }
```

And here I don' remember exactly, but you definitely should be able to use your enum values in queries either by one of this two variants
```ruby
scope :by_requested, -> { where(stage: :requested) }
# OR
scope :by_requested, -> { where(stage: DriverRequest.stages[:requested]) }
```

File: app/models/user.rb [hunk 2]
```diff
@@ -35,4 +36,14 @@ def age
...
+
+  def jwt_payload
+    {
+      "context": {
+        "id": id,
+        "name": name,
+        "role": role,
+      },
+    }
+  end
```
Comment (yauhenininjia) [inline app/models/user.rb:48]: This is something not related to model and should be moved out

File: app/uploaders/driver_id_uploader.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+class DriverIdUploader < CarrierWave::Uploader::Base
```
Comment (yauhenininjia) [inline app/uploaders/driver_id_uploader.rb:1]: Is there any reason to use Carrierwave for new project instead of ActiveStorage built in Rails?

**Review by yauhenininjia — APPROVED** [review]

File: app/controllers/stats_controller.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class StatsController < ApplicationController
   before_action :only => [:index] do
-    redirect_to root_path unless current_user && current_user.admin?
+    redirect_to root_path unless current_user.admin?
```
[the remark this answers is not in the document]
Comment (danube20) [reply app/controllers/stats_controller.rb:3]: I'll take a look

File: app/models/user.rb (hunk 2 above)
Comment (danube20) [reply app/models/user.rb:48]: According to the tutorial I followed, if I wanted to add more things to the payload, I needed to declare the method in the user model and return a hash with the new values I wanted to add

File: app/uploaders/driver_id_uploader.rb (hunk 3 above)
Comment (danube20) [reply app/uploaders/driver_id_uploader.rb:1]: Andrea wanted me to use it

File: app/models/driver_request.rb (hunk 1 above)
Comment (danube20) [reply app/models/driver_request.rb:14]: I was able to use enum values by calling the value as a symbol!

File: app/jobs/notify_admin_driver_request_job.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class NotifyAdminDriverRequestJob < ApplicationJob
+  queue_as :default
+
+  def perform(id)
```
[the remark this answers is not in the document]
Comment (danube20) [reply app/jobs/notify_admin_driver_request_job.rb:4]: I'll investigate this 🤔

**Pull request merged**